streams/test-utils/src/test/java/org/apache/kafka/streams/TestTopicsTest.java

```diff
@@ -0,0 +1,414 @@
+import static org.hamcrest.CoreMatchers.hasItems;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.hasProperty;
+import static org.junit.Assert.assertThrows;
+
+public class TestTopicsTest {
+    private static final Logger log = LoggerFactory.getLogger(TestTopicsTest.class);
+
+    private final static String INPUT_TOPIC = "input";
+    private final static String OUTPUT_TOPIC = "output1";
+    private final static String INPUT_TOPIC_MAP = OUTPUT_TOPIC;
+    private final static String OUTPUT_TOPIC_MAP = "output2";
+
+    private TopologyTestDriver testDriver;
+    private final Serde<String> stringSerde = new Serdes.StringSerde();
+    private final Serde<Long> longSerde = new Serdes.LongSerde();
+
+    private final Properties config = mkProperties(mkMap(
+            mkEntry(StreamsConfig.APPLICATION_ID_CONFIG, "TestTopicsTest"),
+            mkEntry(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, "dummy:1234")
+    ));
+    private final Instant testBaseTime = Instant.parse("2019-06-01T10:00:00Z");
+
+    @Before
+    public void setup() {
+        final StreamsBuilder builder = new StreamsBuilder();
+        //Create Actual Stream Processing pipeline
+        builder.stream(INPUT_TOPIC).to(OUTPUT_TOPIC);
+        final KStream<Long, String> source = builder.stream(INPUT_TOPIC_MAP, Consumed.with(longSerde, stringSerde));
+        final KStream<String, Long> mapped = source.map((key, value) -> new KeyValue<>(value, key));
+        mapped.to(OUTPUT_TOPIC_MAP, Produced.with(stringSerde, longSerde));
+        testDriver = new TopologyTestDriver(builder.build(), config);
+    }
+
+    @After
+    public void tearDown() {
+        try {
+            testDriver.close();
+        } catch (final RuntimeException e) {
+            // https://issues.apache.org/jira/browse/KAFKA-6647 causes exception when executed in Windows, ignoring it
+            // Logged stacktrace cannot be avoided
+            log.warn("Ignoring exception, test failing in Windows due this exception: {}", e.getLocalizedMessage());
+        }
+    }
+
+    @Test
+    public void testValue() {
+        final TestInputTopic<String, String> inputTopic = testDriver.createInputTopic(INPUT_TOPIC, stringSerde.serializer(), stringSerde.serializer());
+        final TestOutputTopic<String, String> outputTopic = testDriver.createOutputTopic(OUTPUT_TOPIC, stringSerde.deserializer(), stringSerde.deserializer());
+        //Feed word "Hello" to inputTopic and no kafka key, timestamp is irrelevant in this case
+        inputTopic.pipeInput("Hello");
+        assertThat(outputTopic.readValue(), equalTo("Hello"));
+        //No more output in topic
+        assertThat(outputTopic.isEmpty(), is(true));
+    }
+
+    @Test
+    public void testValueList() {
+        //Note using here string key serde even other topic is expecting long
```

Comment:
Do we still need this comment? Seems to me the other topic is expecting a `String` as well, but I could be missing something.

Reply:
Removed